### ISSUE TITLE
[FR] Add command for "Copy selection to chat input"

### ISSUE THREAD

**Issue by logancyang** [logancyang/obsidian-copilot#751]
Reference: https://github.com/logancyang/obsidian-copilot/discussions/746

Whether to add it to Obsidian's "context menu", or as a command, or both, WDYT? @MTGMAD

I don't know if you can set keyboard shortcuts for context menu items, but for command yes.

**Comment by HaenirFae** [comment]
Would love to see this feature added.

**Comment by logancyang** [comment]
We will have a more streamlined experience led by @zeroliu that consolidates inline commands, custom prompts into a "cmd + K like" experience.

**Comment by AgatElite** [comment]
Is there a plan to add this to the free version of the plugin by any chance?

**Comment by zeroliu** [comment]
I believe this is available to the free version

**Comment by zeroliu** [comment]
@AgatElite we have launched the feature in case you missed it

**Comment by AgatElite** [comment]
I'm not seeing it, is it: 'Add selection to chat context'? if so, it's not in the free version, I believe.

**Comment by zeroliu** [comment]
It's called "Quick command". Where you can highlight texts and trigger the quick command to trigger AI response in the editor.

**Comment by AgatElite** [comment]
> It's called "Quick command". Where you can highlight texts and trigger the quick command to trigger AI response in the editor.

Yes, I know the feature you're referring to, I've played around with it, but I found myself going back to copy-pasting text into the sidebar chat and proceeding from there.
Having asked myself why, these are the reasons:

1. Answers don't get rendered like they do in copilot chat, so for example any latex contained in the response doesn't get rendered.
2. It doesn't allow you to view the previous response you received about your selection. This, for me, makes it harder to have a backwards and forwards conversation in the context of my selection.

**Comment by zeroliu** [comment]
> > It's called "Quick command". Where you can highlight texts and trigger the quick command to trigger AI response in the editor.
> 
> Yes, I know the feature you're referring to, I've played around with it, but I found myself going back to copy-pasting text into the sidebar chat and proceeding from there. Having asked myself why, these are the reasons:
> 
> 1. Answers don't get rendered like they do in copilot chat, so for example any latex contained in the response doesn't get rendered.
> 2. It doesn't allow you to view the previous response you received about your selection. This, for me, makes it harder to have a backwards and forwards conversation in the context of my selection.

Thanks for clarifying. Quick Command was designed as a lightweight way to trigger Copilot without interrupting your ongoing chat or requiring the chat interface. For your workflow, “Add selection to chat context” is probably more suitable. I’ll discuss with @logancyang whether we can make this available to free users. Since we recently expanded context management to all free users, this is a feature we could consider as well.

**Comment by AgatElite** [comment]
Any update on whether this feature is being considered or not for the free version?

**Comment by logancyang** [comment]
@AgatElite This has already been added several versions ago. Closing.

**Comment by logancyang** [comment]
I stand corrected, this is guarded as a Plus feature now which it shouldn't be since chat context menu is open to free users. Will post a fix.

**Comment by zeroliu** [comment]
@AgatElite I will pick this up later this week

**Comment by zeroliu** [comment]
https://github.com/logancyang/obsidian-copilot/pull/1867
